cdap-app-fabric/src/main/java/io/cdap/cdap/app/preview/PreviewManager.java

```diff
@@ -35,10 +43,40 @@
   ApplicationId start(NamespaceId namespace, AppRequest<?> request) throws Exception;
 
   /**
-   * Get the {@link PreviewRunner} responsible for managing the preview.
-   * @return the {@link PreviewRunner} associated with the preview
+   * Get the status of the preview.
+   * @param applicationId id of the preview application for which preview status is to be returned
+   * @return the status of the preview
+   * @throws NotFoundException if preview application is not found
+   */
+  PreviewStatus getStatus(ApplicationId applicationId) throws NotFoundException;
+
+  /**
+   * Stop the preview run represented by this {@link ApplicationId}.
+   * @param applicationId id of the preview
+   * @throws Exception thrown when any error in stopping the preview run
+   */
+  void stopPreview(ApplicationId applicationId) throws Exception;
+
+  /**
+   * Get the data associated with the preview run represented by this {@link PreviewRunner}.
+   * @param applicationId the id of the preview application
+   * @param tracerName the name of the tracer used for preview
+   * @return the {@link Map} of properties associated with the tracer for a given preview
+   */
+  Map<String, List<JsonElement>> getData(ApplicationId applicationId, String tracerName);
+
+  /**
+   * Get the run record of the program executed as a part of preview.
+   * @param applicationId the id of the preview application
+   * @return the {@link ProgramRunId} associated with the preview or {@code null} if there is no run record
+   */
+  ProgramRunId getRunRecord(ApplicationId applicationId) throws Exception;
```

Comment:
It says getting RunRecord, but what it returns is just the run id. Shall we rename the method?

Reply:
Renamed method.